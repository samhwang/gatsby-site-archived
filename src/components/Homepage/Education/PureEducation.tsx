import React from 'react';
import School from './School';

function PureEducation({ education }: { education: EducationMetadata }) {
  return (
    <section
      className="resume-section p-3 p-lg-5 d-flex align-items-center"
      id="education"
    >
      <div className="w-100">
        <h2 className="mb-5">Education</h2>
        {education.map(({ institute, degree, major, duration }) => (
          <School
            institute={institute}
            degree={degree}
            major={major}
            duration={duration}
            key={institute}
          />
        ))}
      </div>
    </section>
  );
}

export default PureEducation;
