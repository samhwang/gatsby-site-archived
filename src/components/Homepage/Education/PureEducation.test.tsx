/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
import PureEducation from './PureEducation';

describe('Education Component', () => {
  it('Should render without crashing and match snapshot', () => {
    const education = [
      {
        institute: 'Navitas Professionals',
        degree: 'Professional Year Program in Computer Science',
        major: 'incl. Cert IV in Business',
        duration: 'Oct 2016 - Oct 2017',
      },
    ];
    const tree = renderer
      .create(<PureEducation education={education} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
