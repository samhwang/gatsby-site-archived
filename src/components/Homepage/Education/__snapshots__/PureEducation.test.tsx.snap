// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education Component Should render without crashing 1`] = `
<section
  className="resume-section p-3 p-lg-5 d-flex align-items-center"
  id="education"
>
  <div
    className="w-100"
  >
    <h2
      className="mb-5"
    >
      Education
    </h2>
    <div
      className="resume-item d-flex flex-column flex-md-row justify-content-between mb-5"
    >
      <div
        className="resume-content"
      >
        <h3
          className="mb-0"
        >
          Navitas Professionals
        </h3>
        <div
          className="subheading mb-3"
        >
          Professional Year Program in Computer Science
        </div>
        <div>
          incl. Cert IV in Business
        </div>
      </div>
      <div
        className="resume-date text-md-right"
      >
        <span
          className="text-primary"
        >
          Oct 2016 - Oct 2017
        </span>
      </div>
    </div>
  </div>
</section>
`;
